src/mca/prteif/base/if_base_components.c

```diff
@@ -281,35 +293,50 @@ static char **split_and_resolve(char **orig_str, char *name)
-            prte_ifindextoaddr(if_index, (struct sockaddr *) &if_inaddr, sizeof(if_inaddr));
-            if (prte_net_samenetwork(&argv_inaddr, &if_inaddr, argv_prefix)) {
-                break;
+
+        /* Go through all interfaces to see if we can find one or more matches */
+        match_count = 0;
+        for (if_index = prte_ifbegin(); if_index >= 0;
+             if_index = prte_ifnext(if_index)) {
+            prte_ifindextoaddr(if_index,
+                               (struct sockaddr*) &if_inaddr,
+                               sizeof(if_inaddr));
+            if (prte_net_samenetwork((struct sockaddr*) &argv_inaddr,
+                                     (struct sockaddr*) &if_inaddr,
+                                     argv_prefix)) {
+                /* We found a match. If it's not already in the interfaces array,
+                   add it. If it's already in the array, treat it as a match */
+                match_count = match_count + 1;
+                prte_ifindextoname(if_index, if_name, sizeof(if_name));
+                for (n = 0; n < interface_count; n++) {
+                    if (0 == strcmp(if_name, interfaces[n])) {
+                        break;
+                    }
+                }
+                if (n == interface_count) {
+                    prte_output_verbose(20,
+                                        prte_prteif_base_framework.framework_output,
+                                        "prteif:base: Found match: %s (%s)",
+                                        prte_net_get_hostname((struct sockaddr*) &if_inaddr),
+                                        if_name);
+                    prte_argv_append(&interface_count, &interfaces, strdup(if_name));
+                }
             }
         }
+
         /* If we didn't find a match, keep trying */
-        if (if_index < 0) {
+        if (0 == match_count) {
             prte_show_help("help-oob-tcp.txt", "invalid if_inexclude", true, name,
                            prte_process_info.nodename, tmp,
                            "Did not find interface matching this subnet");
             free(tmp);
-            continue;
         }
-
-        /* We found a match; get the name and replace it in the argv */
-        prte_ifindextoname(if_index, if_name, sizeof(if_name));
-        prte_output_verbose(20, prte_prteif_base_framework.framework_output,
-                            "%s if: Found match: %s (%s)", PRTE_NAME_PRINT(PRTE_PROC_MY_NAME),
-                            prte_net_get_hostname((struct sockaddr *) &if_inaddr), if_name);
-        argv[save++] = strdup(if_name);
-        free(tmp);
     }
 
-    /* The list may have been compressed if there were invalid
-     entries, so ensure we end it with a NULL entry */
-    argv[save] = NULL;
+    /* Mark the end of the interface name array with NULL */
+    interfaces[interface_count] = NULL;
+    free(argv);
```

Comment:
Shouldn't this be `prte_argv_free(argv)`?

Reply:
Yes, fixed. Some argv array elements were being freed earlier in the function, so those calls to free are removed.